src/login/http.rs

```diff
@@ -87,23 +89,24 @@ pub async fn http_server_get_params() -> Result<String> {
         async move { Ok::<_, hyper::Error>(server_fn_gen(tx_clone)) }
     });
 
-    let runtime = tokio::runtime::Runtime::new()?;
-    let _handle: tokio::task::JoinHandle<Result<(), anyhow::Error>> = runtime.spawn(async {
+    let _handle: tokio::task::JoinHandle<Result<(), anyhow::Error>> = tokio::task::spawn(async {
         let addr = ([127, 0, 0, 1], 8976).into();
 
         let server = Server::bind(&addr).serve(service);
         match server.await {
             Ok(_) => Ok(()),
-            Err(_) => anyhow::bail!("Server binding failed"),
+            Err(_) => anyhow::bail!(display_error_info("Local HTTP Server binding failed.")),
         }
     });
 
     // Receive authorization code and csrf state from HTTP server
-    let params = runtime.block_on(async {
+    let params = tokio::task::spawn(async move {
         match rx.recv().await {
             Some(values) => values,
             None => "error".to_string(),
```

Comment:
since we have a history of difficult-to-debug channel errors, maybe we could be more descriptive and log this error (unless it is already being logged somewhere in the calling context). 

do you know if it is possible that the `rx` can receive `None` at any point _before_ it gets `Some(values)` and should therefore be polled?

Reply:
Logging is a good idea. The only case that I am aware of is when the sender side is closed for any reason, and `rx` still tries to read something. In this case, it should receive `None`.